lib/renderer/api/desktop-capturer.ts

```diff
@@ -1,12 +1,24 @@
 import { nativeImage } from 'electron'
 import { ipcRendererInternal } from '@electron/internal/renderer/ipc-renderer-internal'
 
+const { hasSwitch } = process.electronBinding('command_line')
+
 // |options.types| can't be empty and must be an array
 function isValid (options: Electron.SourcesOptions) {
   const types = options ? options.types : undefined
   return Array.isArray(types)
 }
 
+const enableStacks = hasSwitch('enable-api-filtering-logging')
+
+function getCurrentStack (caller: Function) {
```

Comment:
I'm not sure that `caller` is the right thing to pass here, as it means the stack trace doesn't include the name of the function of interest. e.g.

```
> function withCaller(caller) { const x = {}; Error.captureStackTrace(x, caller); return x.stack }
> function foo() { console.log(withCaller(foo)) }
> function withSelf() { const x = {}; Error.captureStackTrace(x, withSelf); return x.stack }
> function bar() { console.log(withSelf()) }
> foo()
Error
    at repl:1:1 
    at Script.runInThisContext (vm.js:123:20)
    at REPLServer.defaultEval (repl.js:332:29)
    at bound (domain.js:393:14)
    at REPLServer.runBound [as eval] (domain.js:406:12)
    at REPLServer.onLine (repl.js:642:10)
    at REPLServer.emit (events.js:202:15)
    at REPLServer.EventEmitter.emit (domain.js:439:20)
    at REPLServer.Interface._onLine (readline.js:308:10)
    at REPLServer.Interface._line (readline.js:654:8)
undefined
> bar()
Error
    at bar (repl:1:30)
    at repl:1:1
    at Script.runInThisContext (vm.js:123:20)
    at REPLServer.defaultEval (repl.js:332:29)
    at bound (domain.js:393:14)
    at REPLServer.runBound [as eval] (domain.js:406:12)
    at REPLServer.onLine (repl.js:642:10)
    at REPLServer.emit (events.js:202:15)
    at REPLServer.EventEmitter.emit (domain.js:439:20)
    at REPLServer.Interface._onLine (readline.js:308:10)
undefined
```

calling `bar()` lists `bar` at the top of the stack trace, but `foo()` doesn't.

Reply:
it shows
```
WebContents (1): remote.require('fs') Error
    at foobar (/private/var/folders/8r/wrg27t2s7dj035jmgf4kwqvm0000gn/T/tmp-46573iyoYocpIXfBO/renderer.js:5:8)
    at Object.<anonymous> (/private/var/folders/8r/wrg27t2s7dj035jmgf4kwqvm0000gn/T/tmp-46573iyoYocpIXfBO/renderer.js:8:1)
    at Object.<anonymous> (/private/var/folders/8r/wrg27t2s7dj035jmgf4kwqvm0000gn/T/tmp-46573iyoYocpIXfBO/renderer.js:10:3)
    at Module._compile (internal/modules/cjs/loader.js:948:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:960:10)
    at Module.load (internal/modules/cjs/loader.js:794:32)
    at Module._load (internal/modules/cjs/loader.js:707:12)
    at Module._load (electron/js2c/asar.js:732:26)
    at Function.Module._load (electron/js2c/asar.js:732:26)
    at Object.<anonymous> (electron/js2c/renderer_init.js:2382:16)
```

Reply:
otherwise it would be
```
WebContents (1): remote.require('fs') Error
    at Object../lib/renderer/api/remote.js.exports.require (electron/js2c/renderer_init.js:1535:73)
    at foobar (/private/var/folders/8r/wrg27t2s7dj035jmgf4kwqvm0000gn/T/tmp-46573pedsayHGW7Yr/renderer.js:5:8)
    at Object.<anonymous> (/private/var/folders/8r/wrg27t2s7dj035jmgf4kwqvm0000gn/T/tmp-46573pedsayHGW7Yr/renderer.js:8:1)
    at Object.<anonymous> (/private/var/folders/8r/wrg27t2s7dj035jmgf4kwqvm0000gn/T/tmp-46573pedsayHGW7Yr/renderer.js:10:3)
    at Module._compile (internal/modules/cjs/loader.js:948:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:960:10)
    at Module.load (internal/modules/cjs/loader.js:794:32)
    at Module._load (internal/modules/cjs/loader.js:707:12)
    at Module._load (electron/js2c/asar.js:715:26)
    at Function.Module._load (electron/js2c/asar.js:715:26)
```

Reply:
maybe it's better like even even though the function name is both in the log message and the stack trace